contracts/ShortNameClaims.sol

```diff
@@ -32,24 +32,48 @@ contract ShortNameClaims is Ownable {
     using BytesUtils for bytes;
     using StringUtils for string;
 
+    enum Phase {
+        OPEN,
+        REVIEW,
+        FINAL
+    }
+
+    enum Status {
+        PENDING,
+        APPROVED,
+        DECLINED,
+        WITHDRAWN
+    }
+
     struct Claim {
         bytes32 labelHash;
         address claimant;
         uint paid;
+        Status status;
     }
 
     PriceOracle public priceOracle;
     BaseRegistrar public registrar;
     mapping(bytes32=>Claim) public claims;
-    uint public claimCount;
+    mapping(bytes32=>bool) approvedNames;
+    uint public pendingClaims;
+    uint public unresolvedClaims;
+    Phase public phase;
+    address public ratifier;
 
-    event ClaimSubmitted(string claimed, bytes dnsname, uint paid, address claimant);
-    event ClaimApproved(bytes32 indexed claimId);
-    event ClaimDeclined(bytes32 indexed claimId);
+    event ClaimSubmitted(string claimed, bytes dnsname, uint paid, address claimant, string email);
+    event ClaimStatusChanged(bytes32 indexed claimId, Status status);
 
-    constructor(PriceOracle _priceOracle, BaseRegistrar _registrar) public {
+    constructor(PriceOracle _priceOracle, BaseRegistrar _registrar, address _ratifier) public {
         priceOracle = _priceOracle;
         registrar = _registrar;
+        ratifier = _ratifier;
```

Comment:
Is it worth adding the `require` statement that ratifier cannot be nil nor same as the owner as there are some code treating that owner and ratifier addresses are different?

Reply:
I don't think it's a problem if the ratifier is the same as the owner in a given deployment.